### ISSUE TITLE
Backup fails: mysqldump deprecated

### ISSUE THREAD

**Issue by Backroads4Me** [BookStackApp/BookStack#5373]
### Describe the Bug

Use the system cli to execute a backup.   I have historically done this with a cron job.   For example:
/app/www/bookstack-system-cli backup /backups/bookstack.zip

Now the process fails due to mysqldump being deprecated on maria-db for mariadb-dump

Error received:
Checking system requirements...
Dumping the database via mysqldump...

An error occurred when attempting to run a command:

Failed mysqldump with errors:
/usr/bin/mysqldump: Deprecated program name. It will be removed in a future release, use '/usr/bin/mariadb-dump' instead

### Steps to Reproduce

Use the system cli to execute a backup. 
For example:  /app/www/bookstack-system-cli backup /backups/bookstack.zip

### Expected Behaviour

bookstack.zip created

### Exact BookStack Version

v24.12

**Comment by ssddanbrown** [comment]
Hi @Backroads4Me, Thanks for reporting.
I assume this is within the linuxserver container based upon the commands?

We'll probably need to add specific checks for `mariadb-dump` and use that instead as a higher priority default.

**Comment by Backroads4Me** [comment]
That's correct, and it uses image: lscr.io/linuxserver/mariadb
My full uses case is using image: offen/docker-volume-backup to automatically backup Bookstack.   Bookstack has historically ran the backup script daily and dumped the zip file to a bound volume then docker-volume-backup exports the zip file to Amazon S3.

**Comment by lennart-m** [comment]
Same issue for me (using linuxserver.io). Are there plans to fix this? I haven't been able to backup my instance since december.

Thanks a lot!

**Comment by ssddanbrown** [comment]
I have now updated the CLI to check for mariadb-dump before defaulting to mysqldump.
The updated CLI will be part of the next patch release, or otherwise [can be downloaded here](https://github.com/BookStackApp/BookStack/raw/94b1cffa2d57141185d42965c8fa9f2da07f30ae/bookstack-system-cli).

Note: Environments using mariadb clients (including mariadb-dump) should also use mariadb as their database server. During testing I've found compatibility is starting to get trickier as those database diverge.

**Comment by hexabinaer** [comment]
Does that mean this page is outdated? https://www.bookstackapp.com/docs/admin/backup-restore/

**Comment by ssddanbrown** [comment]
@hexabinaer No, those fundemental steps are still relevant. It's just that our CLI essentially automates those steps. 
I don't mention the CLI there yet since I've had little feedback, but maybe I should, just with a little warning.
